qiskit/circuit/library/graph_state.py

```diff
@@ -10,13 +10,14 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Graph State circuit."""
+"""Graph State circuit and gate."""
 
 from __future__ import annotations
 
 import numpy as np
-from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.quantumcircuit import QuantumCircuit, Gate
 from qiskit.circuit.exceptions import CircuitError
+from qiskit.utils.deprecation import deprecate_func
 
 
 class GraphState(QuantumCircuit):
```

Comment:
Can we make this circuit use the gate internally?

Reply:
Great suggestion! Done.